### PR TITLE
Throw NotSupportedException for WaitHandle.WaitAny with 64 handle on STA threads

### LINKED ISSUE #1646: WaitHandle.WaitAny with 64 handle on STA threads
Description of the breaking change
---

Prior to .NET 5, passing in 64 handles to WaitAny on an STA thread incorrectly returned a timeout error code immediately. .NET 5+ throws NotSupportedException in this case.

Repro
---

```csharp
using System;
using System.Threading;

class Program
{
    static void Main(string[] args)
    {
        var mres = new ManualResetEventSlim();
        var t = new Thread(() =>
        {
            var handles = new WaitHandle[64];
            for (int i = 0; i < handles.Length; i++)
            {
                handles[i] = new ManualResetEvent(false);
            }
            Console.WriteLine("waiting");
            try
            {
                int i = WaitHandle.WaitAny(handles);
                Console.WriteLine(i);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
            }
            finally
            {
                mres.Set();
            }
        });
        t.SetApartmentState(ApartmentState.STA);
        t.IsBackground = true;
        t.Start();
        mres.Wait();
    }
}
```

Expected behavior: Exception or hang
Actual behavior: 258 (WaitTimeout)

More context https://github.com/dotnet/runtime/pull/636#issuecomment-573359886

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #1646

### CODE REVIEW

**Comment by jkotas** [comment]
This is technically a breaking change, but I believe it is better to throw than silently do the wrong thing. It is unlikely that a working program depends on the current broken behavior.

If we agree that that this is a good change to make, I am also going to send PR to update the docs.

File: src/libraries/System.Runtime/tests/System/Threading/WaitHandleTests.cs [hunk 1]
```diff
@@ -78,14 +78,39 @@ public static void WaitAny_SameHandles()
             Assert.Equal(0, WaitHandle.WaitAny(wh));
         }
 
+        static ManualResetEvent[] CreateManualResetEvents(int length)
+        {
+            var handles = new ManualResetEvent[63];
```
Comment (stephentoub) [inline src/libraries/System.Runtime/tests/System/Threading/WaitHandleTests.cs]:
```suggestion
            var handles = new ManualResetEvent[length];
```

**Review by stephentoub — APPROVED** [review]
I was surprised by the lack of exception. I think this is worth changing.

**Comment by NN---** [comment]
Why breaking ?
On some implementations, if more that 64 handles are passed, a NotSupportedException is thrown.

https://docs.microsoft.com/en-us/dotnet/api/system.threading.waithandle.waitany?view=netframework-4.8#System_Threading_WaitHandle_WaitAny_System_Threading_WaitHandle___

**Comment by stephentoub** [comment]
> Why breaking ?

On an STA thread, passing in 64 handles to WaitAny currently results in it returning a timeout error code. This change makes it throw.

**Comment by NN---** [comment]
Is it documented somewhere?

**Comment by jkotas** [comment]
> Is it documented somewhere?

It is not that's part of the problem. I am updating the documentation in https://github.com/dotnet/dotnet-api-docs/pull/3767

File: src/libraries/System.Runtime/tests/System/Threading/WaitHandleTests.cs [hunk 2]
```diff
@@ -78,14 +78,39 @@ public static void WaitAny_SameHandles()
...
+
         [Fact]
         public static void WaitAll()
         {
-            var handles = new ManualResetEvent[] {
-            new ManualResetEvent(true),
-            new ManualResetEvent(true),
-            new ManualResetEvent(true)
-        };
+            ManualResetEvent[] handles = CreateManualResetEvents(3);
```
Comment (danmosemsft) [inline src/libraries/System.Runtime/tests/System/Threading/WaitHandleTests.cs]: I guess disposing these isn’t important

**Pull request merged**